Pocket/pocket-ios
# Highlights - Highlights filter and highlights list with share and delete in article view

## Summary
This PR:
* Adds the Article Highlights List View to the Article view. 
    - The list can be accessed from the overflow menu. 
    - From this list, you can share or delete an highlight by tapping on the corresponding button
    - Tapping on an highlight, will scroll to the highlight position
* Adds the Highlights filter to saved and archived lists

> [!NOTE]
> * accessing the list by tapping an highlight will be implemented in a future PR
> * Add highlights by selecting text will be added in a future PR
## References 
* Issue number in branch name

## Implementation Details
* Add `HighlightedQuote`, an `Identifiable` type that will be the model in the Highlights list view
* Refactor highlighting logic in `ArticleComponentHighlights` to attach highlight indexes to each component. These indexes are the ones of the sorted `Highlight` array that comes from Core Data, and are used to reference quotes (to display) and Identifiers (when a user removes an existing highlights). The logic now distinguishes between the start of an highlight and the continuation, that is an highlight that spans multiple components and started in a previous component. This is done to preserve original highlights that might span more than one component.
* Update `ArticleComponentPresenter`(s), to carry information about highlights associated to that component, so that we can construct the `highlightedQuotes` list, to feed the Article Highlights view
* Update `SavedItemViewModel`, move highlights sorting logic into this class from `ArticleComponentHighlights`
* Update `ReadableViewController`, add `highlightQuotes` property and logic to populate it.

## Test Steps
* Build/run this branch
* Go to Saves then tap the "Highlights" filter
* Tap the overflow menu
* Make sure the "Highlights" Item appears
* Tap on it and make sure the highlight lists appears in a modal
* Make sure delete and share behave as expected

## PR Checklist:

## Video

https://github.com/Pocket/pocket-ios/assets/34376330/196093c1-08d4-4ae4-99cc-6308aba72108
